chia-bls/src/signature.rs

```diff
@@ -11,14 +14,30 @@ use std::io::Cursor;
 use std::mem::MaybeUninit;
 use std::ops::{Add, AddAssign};
 
+#[cfg(feature = "py-bindings")]
+use crate::public_key::parse_hex_string;
+#[cfg(feature = "py-bindings")]
+use chia_py_streamable_macro::PyStreamable;
+#[cfg(feature = "py-bindings")]
+use chia_traits::from_json_dict::FromJsonDict;
+#[cfg(feature = "py-bindings")]
+use chia_traits::to_json_dict::ToJsonDict;
+#[cfg(feature = "py-bindings")]
+use pyo3::{pyclass, pymethods, IntoPy, PyAny, PyObject, PyResult, Python};
+
 // we use the augmented scheme
 pub const DST: &[u8] = b"BLS_SIG_BLS12381G2_XMD:SHA-256_SSWU_RO_AUG_";
 
+#[cfg_attr(
+    feature = "py-bindings",
+    pyclass(name = "G2Element"),
+    derive(PyStreamable)
+)]
 #[derive(Clone)]
 pub struct Signature(pub(crate) blst_p2);
 
 impl Signature {
-    pub fn from_bytes(buf: &[u8; 96]) -> Result<Self> {
+    pub fn from_bytes_unchecked(buf: &[u8; 96]) -> Result<Self> {
```

Comment:
Being named `_unchecked` implies that this function is `unsafe` and that there is a safe version that performs such checks. If this function can cause undefined behavior it should be marked `unsafe`. It also looks like it returns a `Result`, which doesn't make sense if it's unchecked.

Reply:
this is the current naming convention in `blspy`. It means the G2 point is not validated to be on the G2 curve. This check is quite expensive. When serializing and deserializing between two trusted party we try to avoid it. But when reading an untrusted signature, it's critical to check it, as the signature verification may give false positives otherwise (at least that's my understanding)